### ISSUE TITLE
Non portable key symbols

### ISSUE THREAD

**Issue by jonenst** [factor/factor#1656]
For example, on windows, you can never send the "ENTER" key symbol that other platforms send.
(The difference between "RET" and "ENTER" can be seen in the following picture: https://upload.wikimedia.org/wikipedia/commons/a/a5/Enter.png)

Other key symbols are non portable. Here are the lists: 
non portable:

missing from windows:
- ENTER

missing from mac:
- F9
- F10
- F11
- F12

missing from x11:
- F10
- F11
- F12

only on windows:
- INSERT

only on mac:
- CLEAR

only on linux (x11 or gtk):
- BEGIN

For reference, here's the list of portable key symbols:

```
"BACKSPACE"
"DELETE"
"DOWN"
"END"
"ESC"
"F1"
"F2"
"F3"
"F4"
"F5"
"F6"
"F7"
"F8"
"HOME"
"LEFT"
"PAGE_DOWN"
"PAGE_UP"
"RET"
"RIGHT"
"TAB"
"UP"
```

**Comment by bjourne** [comment]
I see that `INSERT` was missing on gtk, but I'm not so sure about the other keys. F10, F11 and F12 seem to work fine on my keyboard?

**Comment by jonenst** [comment]
F10-12 work with the x11 backend? Maybe I misread the code.

**Comment by bjourne** [comment]
You are right. I didn't realize the gtk and x11 backend handled keys separately.

**Comment by bjourne** [comment]
The latest commit should take care of the remaining x11 vs gtk differences.

**Comment by mrjbq7** [comment]
I'm confused about non-portable. Does that mean our backends don't have those symbols mapped?  Any reason not to apply some patches like this one:

```diff
diff --git a/basis/ui/backend/x11/keys/keys.factor b/basis/ui/backend/x11/keys/keys.factor
index 5184a32..a78ead5 100644
--- a/basis/ui/backend/x11/keys/keys.factor
+++ b/basis/ui/backend/x11/keys/keys.factor
@@ -29,6 +29,7 @@ CONSTANT: codes
         { $ XK_Page_Down "PAGE_DOWN" }
         { $ XK_End "END" }
         { $ XK_Begin "BEGIN" }
+        { $ XK_Insert "INSERT" }
         { $ XK_F1 "F1" }
         { $ XK_F2 "F2" }
         { $ XK_F3 "F3" }
```

**Comment by bjourne** [comment]
I fixed that a while ago. :) There may be some remaining differences on Windows and OS X still but not on Linux (afaik).

**Comment by mrjbq7** [comment]
Haha so you did. That's funny. I just looked at one spot and didn't `grep`.
